usagov/vote-gov-drupal
# VOTE-1350 Enable and configure config split

## Jira ticket

[VOTE-1350](https://bixal-projects.atlassian.net/browse/VOTE-1350) - Enable Config Split
[VOTE-1351](https://bixal-projects.atlassian.net/browse/VOTE-1351) - Add Config Split settings to settings.php file

## Description

We are adding and configuring the Config Split module for local, non production, and production environments. 
Additionally we are adding the Devel module to use on local environments.

## Deployment and testing

### Pre-deploy

1. None

### Post-deploy

1. Local environment will require you to add the following lines to your settings.local.php file (you can also reference .lando/settings.local.php)
`$settings['config_sync_directory'] = dirname(DRUPAL_ROOT) . '/config/sync';
$config['config_split.config_split.local']['status'] = TRUE;`
2. Execute `lando retune` to import the configurations for the local environment (This will be done automatically for cloud environments via the pipeline)

### QA/Test
Follow the following steps after confirming that `drush cim` was successfully executed.
1. Visit /admin/config/development/configuration/config-split and confirm that the correct config split setting is active.
2. Visit /admin/config/development/configuration and confirm that the message, "The staged configuration is identical to the active configuration.", is displayed.
3. Confirm that the following modules are enabled, disabled, or configured given the following requirements:

**Local**
- Enable Devel
- Enable DB Log
- Autologout - has increased logout time
- Disable External Auth
- Disable log_stdout
- Disable New Relic
- Disable Usagov Login
- Disable S3FS

**Non-production**
- Enable External Auth
- Enable New Relic
- Enable Usagov Login
- Enable DB Log
- Enable S3FS
- Disable log_stdout

**Production**
- Enable External Auth
- Enable log_stdout
- Enable New Relic
- Enable Usagov Login
- Enable S3FS
- Disable DB Log
- Disable Devel

## Checklist for the Developer



- [x] This work meets the project [standards](/docs/standards.md).

## Checklist for the Peer Reviewers
